packages/x-date-pickers-pro/src/MultiInputDateRangeField/useMultiInputDateRangeField.ts

```diff
@@ -36,16 +34,10 @@ export const useMultiInputDateRangeField = <
     }
 
     return (newDate: TDate | null) => {
-      let newDateRange: DateRange<TDate>;
-      if (valueProp !== undefined) {
-        newDateRange = dateRangePickerValueManager.parseInput(utils, valueProp);
-      } else if (firstDefaultValue.current !== undefined) {
-        newDateRange = dateRangePickerValueManager.parseInput(utils, firstDefaultValue.current);
-      } else {
-        newDateRange = dateRangePickerValueManager.emptyValue;
-      }
-
-      newDateRange[index] = newDate;
+      const currentDateRange =
+        valueProp ?? firstDefaultValue.current ?? dateRangePickerValueManager.emptyValue;
```

Comment:
Don't you need to clean this value?

```suggestion
        dateRangePickerValueManager.cleanValue(valueProp) ?? firstDefaultValue.current ?? dateRangePickerValueManager.emptyValue;
```

Reply:
Indeed
I will improve it in a follow up to use `useControlled` and avoid these fallback.